### PR TITLE
http: removing envoy.reloadable features.always apply route header rules

### LINKED ISSUE #16002: envoy.reloadable_features.always_apply_route_header_rules deprecation
Your change #13204 (http: applying route header rules on local replies.) introduced a runtime guarded feature. It has been 6 months since the new code has been exercised by default, so it's time to remove the old code path. This issue tracks source code cleanup so we don't forget.

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: Low
Testing: n/a
Docs Changes: n/a
Release Notes: inline
Fixes #16002

### CODE REVIEW

**Review by yanavlasov — APPROVED** [review]

File: source/common/http/filter_manager.cc [hunk 1]
```diff
@@ -889,9 +889,7 @@ void FilterManager::sendLocalReplyViaFilterChain(
       state_.destroyed_,
       Utility::EncodeFunctions{
           [this, modify_headers](ResponseHeaderMap& headers) -> void {
-            if (streamInfo().route_entry_ &&
-                Runtime::runtimeFeatureEnabled(
-                    "envoy.reloadable_features.always_apply_route_header_rules")) {
+            if (streamInfo().route_entry_) {
```
Comment (asraa) [inline source/common/http/filter_manager.cc:892]: drive-by: might want to remove the runtime dep/include

**Review by yanavlasov — APPROVED** [review]
